### PR TITLE
[WIP] Multi namespace support

### PR DESCRIPTION
**Pull request by melkio**
I'm still working on it, but @Particular/azure-maintainers can start going through the code.

I'm introducing some unit tests to be sure not to "create" regression. 
Then, as discussed on slack, I'm reorganizing namespaces and project structure in general as for ASB transport. Also the config API has been changed to look more similar to ASB config API

I found a possible issue creating queues: current implementation creates only queues listed into `QueueBindings.ReceivingAddresses` . Going through the ASB transport codebase also `QueueBindings.SendingAddresses` (`error`, `audit`,...) should be created if missing. This point was confirmed by @yvesgoeleven too. I'm fixing it.

I'll update PR description during my work to make changes I'll do more evident.

Connects to https://github.com/Particular/NServiceBus.AzureStorageQueues/issues/21

### CODE REVIEW

File: src/Transport/Config/AzureStorageQueueInfrastructure.cs [hunk 1]
```diff
@@ -85,7 +85,7 @@ static MessageWrapperSerializer BuildSerializer(ReadOnlySettings settings)
                 if (serializer == null)
                 {
                     throw new ConfigurationErrorsException($"The bus is configured using different {typeof(SerializationDefinition).Name} than defaults provided by the NServiceBus. " +
-                                                           $"Register a custom serialization with {typeof(AzureStorageTransportExtensions).Name}.SerializeMessageWrapperWith()");
+                                                           $"Register a custom serialization with {typeof(AzureStorageQueueTransportExtensions).Name}.SerializeMessageWrapperWith()");
```
Comment (SeanFeldman) [inline src/Transport/Config/AzureStorageQueueInfrastructure.cs]: > "than defaults provided"

Should it be singular?

File: src/Transport/Config/AzureStorageQueueTransport.cs [hunk 2]
```diff
@@ -1,94 +1,34 @@
...
-            }
-
-            return connectionString;
-        }
-
-        protected override bool RequiresConnectionString
+        void RegisterConnectionStringAsStorageAccount(SettingsHolder settings, string connectionString)
         {
-            get { return false; }
+            var extensions = new AzureStorageQueueAccountPartitioningSettings(settings);
```
Comment (SeanFeldman) [inline src/Transport/Config/AzureStorageQueueTransport.cs]: Is it truly an "extensions" or something more specific?

File: src/Transport/Config/ExtensionPoint/AzureStorageAddressingSettings.cs [hunk 3]
```diff
@@ -2,10 +2,9 @@
 {
     using System;
     using System.Collections.Generic;
-    using NServiceBus.Azure.Transports.WindowsAzureStorageQueues;
-    using NServiceBus.Azure.Transports.WindowsAzureStorageQueues.Config;
+    using NServiceBus.AzureStorageQueues;
 
-    public sealed class AzureStorageAddressingSettings : IAzureStoragePartitioningSettings
+    public sealed class AzureStorageAddressingSettings
```
Comment (SeanFeldman) [inline src/Transport/Config/ExtensionPoint/AzureStorageAddressingSettings.cs]: Should be internalized?

File: src/Transport/Config/WellKnownConfigurationKeys.cs [hunk 4]
```diff
@@ -12,5 +12,14 @@ public static class WellKnownConfigurationKeys
         public const string Sha1Shortener = "Transport.AzureStorageQueue.Sha1Shortener";
         public const string PurgeOnStartup = "Transport.AzureStorageQueue.PurgeOnStartup";
         public const string DefaultQueuePerInstance = "Transport.AzureStorageQueue.DefaultQueuePerInstance";
+
+        public static class Addressing
```
Comment (SeanFeldman) [inline src/Transport/Config/WellKnownConfigurationKeys.cs]: Should configuration key classes be internalized?

**Comment by andreasohlund** [comment]
As discussed with @melkio yesterday, multinamespace is on hold, removing the in-progress label

**Comment by Scooletz** [comment]
@melkio Taking into consideration a vast number of changes applied to the `release-7.0.0` that will lend to `develop` and the fact that the multi account support is on hold, what are your plans with this PR?

**Comment by Scooletz** [comment]
I'm closing this. If we want to move on with multi account support, we should base it on probably on ASB and the current state of ASQ which changed a lot.

**Pull request closed without merging**